extensions/src/main/java/org/pentaho/platform/plugin/services/email/EmailService.java

```diff
@@ -186,18 +207,51 @@ protected PasswordAuthentication getPasswordAuthentication() {
       msg.setText( messages.getString( "EmailService.MESSAGE" ) );
       msg.setHeader( "X-Mailer", "smtpsend" );
       msg.setSentDate( new Date() );
-      Transport.send( msg );
-      sendEmailMessage = "EmailTester.SUCESS";
+      sendEmail( emailConfig, session, msg );
+      sendEmailMessage = TEST_EMAIL_SUCCESS;
     } catch ( Exception e ) {
       logger.error( messages.getString( "EmailService.NOT_CONFIGURED" ), e );
-      sendEmailMessage = "EmailTester.FAIL";
+      sendEmailMessage = TEST_EMAIL_FAIL;
+    }
+    if ( EmailConstants.GRANT_TYPE_AUTH_CODE.equals( emailConfig.getGrantType() ) ) {
+      setEmailConfig( emailConfig );
     }
     return sendEmailMessage;
   }
 
+  public void sendEmail( final IEmailConfiguration emailConfig, Session session, MimeMessage msg ) throws EmailServiceException {
+    try {
+      if ( emailConfig.getAuthMechanism().equals( EmailConstants.AUTH_TYPE_XOAUTH2 ) ) {
+        IEmailAuthenticationResponse token = getOAuthToken( emailConfig );
+        if ( emailConfig.getGrantType().equals( EmailConstants.GRANT_TYPE_AUTH_CODE ) ) {
+          emailConfig.setGrantType( EmailConstants.GRANT_TYPE_REFRESH_TOKEN );
+          emailConfig.setRefreshToken( token.getRefreshToken() );
+          emailConfig.setAuthorizationCode( "" );
+        }
+        if ( emailConfig.getSmtpProtocol().equals( EmailConstants.PROTOCOL_GRAPH_API ) ) {
+          ByteArrayOutputStream os = new ByteArrayOutputStream();
+          msg.writeTo( os );
+          String s = Base64.getEncoder().encodeToString( os.toByteArray() );
+          sendMailGraphApi( emailConfig, token.getAccessToken(), s  );
+        } else {
+          SMTPTransport transport = new SMTPTransport( session, null );
+          transport.connect( emailConfig.getSmtpHost(), emailConfig.getUserId(), null );
+          String userTokenString = String.format( USER_TOKEN_STRING, emailConfig.getUserId(), token.getAccessToken() );
+          String smtpCommand = OAUTH2_COMMAND + Base64Utils.toBase64( userTokenString.getBytes() );
```

Comment:
No further action. Do we need these two base-64 libraries that take a byt[] -> string

line 234 `String s = Base64.getEncoder().encodeToString( os.toByteArray() );`
line 240 `Base64Utils.toBase64( userTokenString.getBytes() );`

I would lean towards Base64 since it's now a standard library in java 11